### PR TITLE
Cursor as context manager

### PR DESCRIPTION
**Pull request by zilder**
Hi!

Here is the patch that adds a context manager interface to the `Cursor` class. So it can be used with `with` statement like:

```
with connection.cursor() as cur:
    cur.execute(query)
```

And another point is that it would be easier to switch from `psycopg2` which implements context manager interface.

### CODE REVIEW

File: pg8000/tests/test_query.py [hunk 1]
```diff
@@ -1,7 +1,7 @@
 import unittest
 import threading
 import pg8000
-from .connection_settings import db_connect
+from pg8000.tests.connection_settings import db_connect
```
Comment (mfenniak) [inline pg8000/tests/test_query.py]: This change doesn't really look correct.  Why was this changed away from a relative import?

**Comment by mfenniak** [comment]
@zilder Thanks for the patch!  I'll try to nurse the CI through here; it seems to be a bit flakey on the last build.  I made a comment on one change that doesn't make sense to me (I suspect it will fail CI in Python 3); I'd appreciate if you could explain that change.

**Comment by zilder** [comment]
> This change doesn't really look correct. Why was this changed away from a relative import?

Sorry about that. I wasn't familiar with the way to run unit tests. I tried to run it just by performing something like:
```
python tests/test_query.py
```
Now I figured it out. By the way, I found the same issue in these tests:
```
test_connection.py
stress.py
performance.py
```
in all cases the author was Tony Locke.

**Comment by mfenniak** [comment]
Thanks for the patch!

**Comment by zilder** [comment]
Thanks!
